packages/yew/src/platform/pinned/mpsc.rs

```diff
@@ -0,0 +1,395 @@
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions and hence uniquely owns the
+        // mutable reference.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+
+        match (inner.items.pop_front(), inner.closed) {
+            (Some(m), _) => Ok(Some(m)),
+            (None, false) => Ok(None),
+            (None, true) => Err(TryRecvError {
+                _marker: PhantomData,
+            }),
+        }
+    }
+}
+
+impl<T> Stream for UnboundedReceiver<T> {
+    type Item = T;
+
+    fn poll_next(
+        self: std::pin::Pin<&mut Self>,
+        cx: &mut std::task::Context<'_>,
+    ) -> std::task::Poll<Option<Self::Item>> {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions and hence uniquely owns the
+        // mutable reference.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+
+        match (inner.items.pop_front(), inner.closed) {
+            (Some(m), _) => Poll::Ready(Some(m)),
+            (None, false) => {
+                inner.rx_waker = Some(cx.waker().clone());
+                Poll::Pending
+            }
+            (None, true) => Poll::Ready(None),
+        }
+    }
+}
+
+impl<T> FusedStream for UnboundedReceiver<T> {
+    fn is_terminated(&self) -> bool {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions and hence uniquely owns the
+        // mutable reference.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &*self.inner.get() };
+        inner.items.is_empty() && inner.closed
+    }
+}
+
+impl<T> Drop for UnboundedReceiver<T> {
+    fn drop(&mut self) {
+        // SAFETY:
+        //
+        // We can acquire a mutable reference without checking as:
+        //
+        // - This type is !Sync and !Send.
+        // - This function is not used by any other functions and hence uniquely owns the
+        // mutable reference.
+        // - The mutable reference is dropped at the end of this function.
+        let inner = unsafe { &mut *self.inner.get() };
+        inner.close();
+    }
+}
+
+/// The sender of an unbounded mpsc channel.
+#[derive(Debug)]
+pub struct UnboundedSender<T> {
+    inner: Rc<UnsafeCell<Inner<T>>>,
+}
+
+impl<T> UnboundedSender<T> {
+    /// Sends a value to the unbounded receiver.
+    pub fn send_now(&self, item: T) -> Result<(), SendError<T>> {
```

Comment:
The `oneshot::Sender` just returns the item if it can't be sent. Why the asymmetric API?

Reply:
This behaviour is consistent with both Tokyo and futures channels.

They have a bound variant, which will emit an error of a reason if either queue being full or channel is closed.